### PR TITLE
Use buildkit for docker builds

### PR DESCRIPTION
**Pull request by rothgar**
**Description of changes:**
Switch from old docker build environment to docker buildkit by default

**Testing done:**
Built the container a few times with and without buildkit. Builds were about 30s faster

Buildkit
real    2m12.879s
user    0m1.077s
sys     0m0.536s

without Buildkit
real    2m45.331s
user    0m0.400s
sys     0m0.294s

### CODE REVIEW

File: Makefile
Comment (samuelkarp) [inline Makefile:40]:
```suggestion
build:
	DOCKER_BUILDKIT=1 docker build $(DOCKER_BUILD_FLAGS) \
```

This looks a bit cleaner to me since we don't need to export `DOCKER_BUILDKIT` for all of the commands in the `build` target (not that there are any others right now...).  Not a strong preference though.

**Review by samuelkarp — APPROVED** [review]

File: Makefile
Comment (rothgar) [reply Makefile:40]: I'm fine with either. The only reason I typically add to all commands is if you end up with multiple docker commands in the build target it can export to all of them at once instead of adding to each.

**Review by webern — APPROVED** [review]

**Comment by tjkirch** [comment]
Hey @rothgar, I wanted to apologize for this PR getting lost and not merged after approval.  Thanks again for your contribution!

**Pull request merged**
